pkg/integrations/v4/dm/id_provider.go

```diff
@@ -4,102 +4,151 @@
 package dm
 
 import (
+	"context"
 	"fmt"
 	"github.com/newrelic/infrastructure-agent/pkg/backend/identityapi"
 	"github.com/newrelic/infrastructure-agent/pkg/entity"
 	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/protocol"
+	"sync"
 )
 
-type RegisteredEntitiesNameToID map[string]entity.ID
-type UnregisteredEntitiesNamed map[string]UnregisteredEntity
+type registeredEntitiesNameToID map[string]entity.ID
+type unregisteredEntitiesNamed map[string]unregisteredEntity
```

Comment:
do we actually need types for those? Why not just use directly the map?

Reply:
It adds some explicitly for our code, you don't need to guess what is a key in the map.